app/helpers/application_helper.rb

```diff
@@ -37,6 +37,12 @@ def user_signing_up?
     !current_user || !current_user.two_factor_enabled?
   end
 
+  def session_with_trust?
+    current_user ||
+      current_page?(controller: 'sign_up/email_confirmations', action: 'create') ||
```

Comment:
sure. the annoying thing about helpers is that they are modules so no great way to make a private method. but I can make a public one :)